examples/graph_animations/example2.md

```diff
@@ -0,0 +1,196 @@
+function dfs_and_animate(node, path)
+    if node==dst
+        highlightNode(GFrames(20+num_visited*10, 100), ag, current, :border_color, "red")
+        return    
+    end
+    # Highlight the current node
+    highlightNode(GFrames(20+num_visited*10, 100), ag, current, :border_color, "yellow")
+    visited[current]=true
+    num_visited+=1
+    push!(path, current)
+    # Change node color when highlighting effect ends
+    changeNodeProperty!(@Frames(prev_end(), stop=parent_end()), ag, current, :color, "blue")
+    for nb in neighbors(g, current)
+        if visited[nb]
+            continue
+        end
+        highlightEdge(GFrames(20+num_visited*10, 100), ag, current, nb, :color, "green")
+        num_visited+=1
+        dfs_and_animate(nb, path)
+    end
+    # Highlight again to indicate return to parent node
+    highlightNode(GFrames(20+num_visited*10, 100), ag, current, :border_color, "yellow")
+    num_visited+=1
+    pop!(path)
+end
+
+dfs_and_animate(1, path)
+```
+
+Note - Even though the drawing property `:color` was not explicitly defined by passing it to a drawing function, these are provided as defaults for every node and edges in the case the graph is created using `create_graph`.
+
+The default drawing properties provided are - 
+* `fill_color` - for nodes
+* `border_color` - for nodes
+* `radius` - for nodes 
+* `color` - for edges
+* `weights` - this property is only available for edges in weighted graphs
+* `opacity`, `scale` & `line_width` - defined in [example 1](example1.md)
+
+If frame management becomes an issue, it is possible to skip it completely and let Javis handle the frame management, again through the use of reasonable defaults. For example, skipping frames in `highlightNode` schedules the node highlighting after the end of the previous animation specified on the graph. This animation could be any of the graph utility functions with the exception of `changeNodeProperty` and `changeEdgeProperty` for which the starting frame is used as reference. To avail of this default option, the `default_keyframes=true` option needs to be passed in `create_graph`.
+
+Most utility functions like `changeNodeProperty` or `highlightNode` or `animate_*` use Javis actions underneath in the implementations which makes it easy to arrange them via frames.
+
+**Using `animate_neighbors`**
+
+Animating neighboring nodes and edges can be simplified by using the `animate_neighbors` utlity function
+
+```julia
+animate_neighbors(ag, 1; animate_node_on=(:fill_color, "green"), animate_edge_on=(:color, "red"))
+```
+This internally makes a call to `changeNodeProperty` and `changeEdgeProperty`. If the keyword arguments are not provided, simple switching highlighting is used for the animation. 
+
+**Using `animate_path`**
+
+After the traversal has been done and the shortest path found, show the path by animating it.
+
+```julia
+# change back the graph to its original form
+changeNodeProperty!(ag, (node)->true, :fill_color, "white")
```

Comment:
the `(node)->true` syntax is quite weird. What does that do?

Reply:
Here we might be able to also specify several changes like above?

Reply:
It is just a function with a fixed signature applying that property change to all nodes matching the criteria.

Reply:
Then maybe make it the default but that's good to know

Reply:
Regarding that you might want to create the function declarations and docstrings next :wink: